### PR TITLE
obs-studio-plugins.obs-multi-rtmp: fix build

### LINKED ISSUE #349178: Build failure: obs-studio-plugins.obs-multi-rtmp
### Steps To Reproduce

1. build `obs-studio-plugins.obs-multi-rtmp`

### Build log

```
obs-multi-rtmp> Running phase: qtPreHook
obs-multi-rtmp> Running phase: unpackPhase
obs-multi-rtmp> unpacking source archive /nix/store/xpv8n36v3gky7h988i6vdjy7brrqq8vv-source
obs-multi-rtmp> source root is source
obs-multi-rtmp> Running phase: patchPhase
obs-multi-rtmp> Running phase: updateAutotoolsGnuConfigScriptsPhase
obs-multi-rtmp> Running phase: configurePhase
obs-multi-rtmp> fixing cmake files...
obs-multi-rtmp> cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/3dqxn67fgq1xmx5dgjjj3ggkw98kgrdd-obs-multi-rtmp-0.6.0.1/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/3dqxn67fgq1xmx5dgjjj3ggkw98kgrdd-obs-multi-rtmp-0.6.0.1/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/3dqxn67fgq1xmx5dgjjj3ggkw98kgrdd-obs-multi-rtmp-0.6.0.1/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/3dqxn67fgq1xmx5dgjjj3ggkw98kgrdd-obs-multi-rtmp-0.6.0.1/share/doc/obs-multi-rtmp -DCMAKE_INSTALL_INFODIR=/nix/store/3dqxn67fgq1xmx5dgjjj3ggkw98kgrdd-obs-multi-rtmp-0.6.0.1/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/3dqxn67fgq1xmx5dgjjj3ggkw98kgrdd-obs-multi-rtmp-0.6.0.1/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/3dqxn67fgq1xmx5dgjjj3ggkw98kgrdd-obs-multi-rtmp-0.6.0.1/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/3dqxn67fgq1xmx5dgjjj3ggkw98kgrdd-obs-multi-rtmp-0.6.0.1/include -DCMAKE_INSTALL_SBINDIR=/nix/store/3dqxn67fgq1xmx5dgjjj3ggkw98kgrdd-obs-multi-rtmp-0.6.0.1/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/3dqxn67fgq1xmx5dgjjj3ggkw98kgrdd-obs-multi-rtmp-0.6.0.1/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/3dqxn67fgq1xmx5dgjjj3ggkw98kgrdd-obs-multi-rtmp-0.6.0.1/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/vh9fsdhgxcnab2qk7vdp2palkkn6j3cp-gcc-wrapper-13.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/vh9fsdhgxcnab2qk7vdp2palkkn6j3cp-gcc-wrapper-13.3.0/bin/ranlib -DCMAKE_AR=/nix/store/vh9fsdhgxcnab2qk7vdp2palkkn6j3cp-gcc-wrapper-13.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/3dqxn67fgq1xmx5dgjjj3ggkw98kgrdd-obs-multi-rtmp-0.6.0.1 -DENABLE_QT:BOOL=TRUE -DENABLE_FRONTEND_API:BOOL=TRUE -DCMAKE_COMPILE_WARNING_AS_ERROR:BOOL=FALSE
obs-multi-rtmp> -- The C compiler identification is GNU 13.3.0
obs-multi-rtmp> -- The CXX compiler identification is GNU 13.3.0
obs-multi-rtmp> -- Detecting C compiler ABI info
obs-multi-rtmp> -- Detecting C compiler ABI info - done
obs-multi-rtmp> -- Check for working C compiler: /nix/store/vh9fsdhgxcnab2qk7vdp2palkkn6j3cp-gcc-wrapper-13.3.0/bin/gcc - skipped
obs-multi-rtmp> -- Detecting C compile features
obs-multi-rtmp> -- Detecting C compile features - done
obs-multi-rtmp> -- Detecting CXX compiler ABI info
obs-multi-rtmp> -- Detecting CXX compiler ABI info - done
obs-multi-rtmp> -- Check for working CXX compiler: /nix/store/vh9fsdhgxcnab2qk7vdp2palkkn6j3cp-gcc-wrapper-13.3.0/bin/g++ - skipped
obs-multi-rtmp> -- Detecting CXX compile features
obs-multi-rtmp> -- Detecting CXX compile features - done
obs-multi-rtmp> -- Found Threads: TRUE
obs-multi-rtmp> -- Performing Test HAVE_STDATOMIC
obs-multi-rtmp> -- Performing Test HAVE_STDATOMIC - Success
obs-multi-rtmp> -- Qt version found: 6
obs-multi-rtmp> -- Found OpenGL: /nix/store/vpys8vaqmjlzcn12wfccym7kgl7b3fr1-libglvnd-1.7.0/lib/libOpenGL.so
obs-multi-rtmp> -- Found WrapOpenGL: TRUE
obs-multi-rtmp> -- Found XKB: /nix/store/0j2mnm88hf6ndp58jybmbiwl4wg492g9-libxkbcommon-1.7.0/lib/libxkbcommon.so (Required is at least version "0.5.0")
obs-multi-rtmp> -- Found WrapVulkanHeaders: /nix/store/jrr54zy7h7w1lhh3rk5q4zcbfwz57r3c-vulkan-headers-1.3.290.0/include
obs-multi-rtmp> CMake Error at cmake/common/helpers_common.cmake:89 (set_property):
obs-multi-rtmp>   set_property can not be used on an ALIAS target.
obs-multi-rtmp> Call Stack (most recent call first):
obs-multi-rtmp>   CMakeLists.txt:25 (find_qt)
obs-multi-rtmp> 
obs-multi-rtmp> -- Configuring incomplete, errors occurred!
error: builder for '/nix/store/mwia79gsmk5j5w782r9hks1y2rvla7d9-obs-multi-rtmp-0.6.0.1.drv' failed with exit code 1;
       last 10 log lines:
       > -- Found WrapOpenGL: TRUE
       > -- Found XKB: /nix/store/0j2mnm88hf6ndp58jybmbiwl4wg492g9-libxkbcommon-1.7.0/lib/libxkbcommon.so (Required is at least version "0.5.0")
       > -- Found WrapVulkanHeaders: /nix/store/jrr54zy7h7w1lhh3rk5q4zcbfwz57r3c-vulkan-headers-1.3.290.0/include
       > CMake Error at cmake/common/helpers_common.cmake:89 (set_property):
       >   set_property can not be used on an ALIAS target.
       > Call Stack (most recent call first):
       >   CMakeLists.txt:25 (find_qt)
       >
       > 
       > -- Configuring incomplete, errors occurred!
       For full logs, run 'nix log /nix/store/mwia79gsmk5j5w782r9hks1y2rvla7d9-obs-multi-rtmp-0.6.0.1.drv'.
```

### Notify maintainers

@06kellyjac 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.111, NixOS, 24.05 (Uakari), 24.05.5366.fbca5e745367`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.8`
 - channels(root): `"nixos-24.05, nixos-hardware, nixos-unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by jackwilsdon**
Fixes #349178.

This has already been changed upstream but no version has been released since: https://github.com/sorayuki/obs-multi-rtmp/compare/c314e604c0d6509fdf047128c9dc2350cd35be4c...a1289fdef404b08a7acbbf0d6d0f93da4c9fc087

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/video/obs-studio/plugins/obs-multi-rtmp/default.nix [hunk 1]
```diff
@@ -11,6 +11,8 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-MRBQY9m6rj8HVdn58mK/Vh07FSm0EglRUaP20P3FFO4=";
   };
 
+  patches = [ ./fix-qt.patch ];
```
Comment (superherointj) [inline pkgs/applications/video/obs-studio/plugins/obs-multi-rtmp/default.nix:14]: Please reference upstream patch by commit using `fetchpatch`.

Like this:
https://github.com/sorayuki/obs-multi-rtmp/commit/a1289fdef404b08a7acbbf0d6d0f93da4c9fc087.patch

Comment (06kellyjac) [reply pkgs/applications/video/obs-studio/plugins/obs-multi-rtmp/default.nix:14]: To clarify, this is what we like to do for merged commits (which this is)

Doing it on in-review commits off a pr is disliked as the commit can disappear when the branch is deleted or force pushed over in review. In that case committing the patch is preferred.

---

If you could also add a comment saying to remove it next release that'd be perfect. Thanks Jack

Comment (jackwilsdon) [reply pkgs/applications/video/obs-studio/plugins/obs-multi-rtmp/default.nix]: It looks like there are no cleanly-applying patches available (seemingly due to this merge: https://github.com/sorayuki/obs-multi-rtmp/commit/33b33a11cb03b74d5c77e0434b440957121b3b0b), so I've had to use `includes` to just patch `CMakeLists.txt` (the patch was failing on `CMakePresets.json`, which conflicted in that merge).

**Review by superherointj — APPROVED** [review]
I confirm this fixes build.

**Review by 06kellyjac — APPROVED** [review]

**Comment by 06kellyjac** [comment]
@NixOS/nixpkgs-merge-bot merge

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**